org.lflang/src/org/lflang/generator/ReactionInstance.java

```diff
@@ -448,6 +448,57 @@ public String toString() {
         return getName() + " of " + parent.getFullName();
     }
 
+    /**
+     * Determine logical execution time for each reaction during compile
+     * time based on immediate downstream logical delays (after delays and actions)
+     * and label each reaction with the minimum of all such delays.
+     */
+    public TimeValue getLogicalExecutionTime() {
+        if (this.let != null) {
+            return this.let;
+        }
+
+        if (this.parent.isGeneratedDelay()) {
+            return this.let = TimeValue.ZERO;
+        }
+
+        TimeValue let = null;
+
+        // Iterate over effect and find minimum delay.
+        for (TriggerInstance<? extends Variable> effect : effects) {
+            if (effect instanceof PortInstance) {
+                var afters = this.parent.getParent().children.stream().filter(c -> {
+                    if (c.isGeneratedDelay()) {
+                        return c.inputs.iterator().next().getDependsOnPorts().iterator().next().instance
```

Comment:
This line is quite hard to understand.

Why are you making calls to `iterator().next()` and not use something like `forEach`?

Reply:
I think I understand this better now based on your previous comment. I would suggest using `.get(0)` if possible to make this clearer.

Reply:
I modified it in https://github.com/lf-lang/lingua-franca/commit/ff0782633922481312131e2321730ef447577194